File: HPWorkpath_v1.6.3/Samples/ExampleAPIServices/source/SuppliesSample/src/com/hp/workpath/sample/suppliessample/view/SupplyListAdapter.java
// Copyright 2018 HP Development Company, L.P.
// SPDX-License-Identifier: MIT
package com.hp.workpath.sample.suppliessample.view;

import android.text.TextUtils;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.hp.workpath.api.supplies.supplyinfo.Supply;
import com.hp.workpath.sample.suppliessample.R;

import java.util.ArrayList;
import java.util.List;

public class SupplyListAdapter extends RecyclerView.Adapter<SupplyListAdapter.ViewHolder> {

    List<Supply> mSupplyList = new ArrayList<>();
    View.OnClickListener mListOnClickListener;

    public SupplyListAdapter(View.OnClickListener listener) {
        mListOnClickListener = listener;
    }

    public Supply getItem(int position) {
        return mSupplyList.get(position);
    }

    public void setItem(List<Supply> supplyList) {
        mSupplyList = supplyList;
        notifyDataSetChanged();
    }

    public void clear() {
        int size = mSupplyList.size();
        if (size > 0) {
            mSupplyList.clear();
            notifyItemRangeRemoved(0, size);
        }
    }

    @Override
    public long getItemId(int position) {
        return position;
    }

    @Override
    public int getItemCount() {
        return mSupplyList.size();
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.layout_supply_list, parent, false);
        view.setOnClickListener(mListOnClickListener);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        Supply supplyData = getItem(position);
        holder.sequenceTextView.setText(String.valueOf(position + 1));
        holder.consumableTypeTextView.setText(supplyData.getConsumableTypeEnum());
        holder.approxPercentTextView.setText(supplyData.getApproxPercentRemaining());
        holder.productNumberTextView.setText(supplyData.getProductNumber());
        holder.descriptionTextView.setText(supplyData.getDescription());
        setDescriptionBackground(holder.descriptionTextView, supplyData.getDescription());
    }

    private void setDescriptionBackground(TextView textView, String description) {
        if (!TextUtils.isEmpty(description)) {
            if (description.contains("Yellow")) {
                textView.setBackgroundResource(R.color.yellow);
            } else if (description.contains("Magenta")) {
                textView.setBackgroundResource(R.color.magenta);
            } else if (description.contains("Cyan")) {
                textView.setBackgroundResource(R.color.cyan);
            } else if (description.contains("Black")) {
                textView.setBackgroundResource(R.color.black);
            }
        }
    }

    public class ViewHolder extends RecyclerView.ViewHolder {
        public TextView sequenceTextView;
        public TextView consumableTypeTextView;
        public TextView consumableStateTextView;
        public TextView approxPercentTextView;
        public TextView productNumberTextView;
        public TextView descriptionTextView;

        ViewHolder(View view) {
            super(view);
            sequenceTextView = view.findViewById(R.id.sequenceTextView);
            consumableTypeTextView = view.findViewById(R.id.consumableTypeTextView);
            approxPercentTextView = view.findViewById(R.id.approxPercentTextView);
            productNumberTextView = view.findViewById(R.id.productNumberTextView);
            descriptionTextView = view.findViewById(R.id.descriptionTextView);
        }
    }
}

File: HPWorkpath_v1.6.3/Samples_Kotlin/ExampleAPIServices/source/SuppliesSample/src/com/hp/workpath/sample/suppliessample/view/SupplyListAdapter.kt
// Copyright 2018 HP Development Company, L.P.
// SPDX-License-Identifier: MIT
package com.hp.workpath.sample.suppliessample.view

import android.text.TextUtils
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.hp.workpath.api.supplies.supplyinfo.Supply
import com.hp.workpath.sample.suppliessample.R
import java.util.ArrayList

class SupplyListAdapter(var mListOnClickListener: View.OnClickListener) : RecyclerView.Adapter<SupplyListAdapter.ViewHolder>() {
    var mSupplyList: MutableList<Supply> = ArrayList()

    private fun getItem(position: Int): Supply {
        return mSupplyList[position]
    }

    fun setItem(supplyList: List<Supply>) {
        mSupplyList = supplyList.toMutableList()
        notifyDataSetChanged()
    }

    fun clear() {
        val size = mSupplyList.size
        if (size > 0) {
            mSupplyList.clear()
            notifyItemRangeRemoved(0, size)
        }
    }

    override fun getItemId(position: Int): Long {
        return position.toLong()
    }

    override fun getItemCount(): Int {
        return mSupplyList.size
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.layout_supply_list, parent, false)
        view.setOnClickListener(mListOnClickListener)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val supplyData = getItem(position)
        holder.sequenceTextView.text = (position + 1).toString()
        holder.consumableTypeTextView.text = supplyData.consumableTypeEnum
        holder.approxPercentTextView.text = supplyData.approxPercentRemaining
        holder.productNumberTextView.text = supplyData.productNumber
        holder.descriptionTextView.text = supplyData.description
        setDescriptionBackground(holder.descriptionTextView, supplyData.description)
    }

    private fun setDescriptionBackground(textView: TextView, description: String) {
        if (!TextUtils.isEmpty(description)) {
            if (description.contains("Yellow")) {
                textView.setBackgroundResource(R.color.yellow)
            } else if (description.contains("Magenta")) {
                textView.setBackgroundResource(R.color.magenta)
            } else if (description.contains("Cyan")) {
                textView.setBackgroundResource(R.color.cyan)
            } else if (description.contains("Black")) {
                textView.setBackgroundResource(R.color.black)
            }
        }
    }

    inner class ViewHolder internal constructor(view: View) : RecyclerView.ViewHolder(view) {
        var sequenceTextView: TextView = view.findViewById(R.id.sequenceTextView)
        var consumableTypeTextView: TextView = view.findViewById(R.id.consumableTypeTextView)
        var approxPercentTextView: TextView = view.findViewById(R.id.approxPercentTextView)
        var productNumberTextView: TextView = view.findViewById(R.id.productNumberTextView)
        var descriptionTextView: TextView = view.findViewById(R.id.descriptionTextView)
    }
}